nodebuilder/p2p/tls.go

```diff
@@ -0,0 +1,51 @@
+package p2p
+
+import (
+	"crypto/tls"
+	"os"
+	"path/filepath"
+
+	"github.com/libp2p/go-libp2p"
+	ws "github.com/libp2p/go-libp2p/p2p/transport/websocket"
+
+	"github.com/celestiaorg/celestia-node/libs/utils"
+)
+
+const (
+	cert = "cert.pem"
+	key  = "key.pem"
+)
+
+var tlsPath = "CELESTIA_TLS_PATH"
+
+// tlsEnabled checks whether `tlsPath` is not empty and creates a certificate.
+// it returns the cfg itself, the bool flag that specifies whether the config was created
+// and an error.
+func tlsEnabled() (*tls.Config, bool, error) {
+	path := os.Getenv(tlsPath)
```

Comment:
check if the string is empty here earlier
and add a debug log if that's the case
this should help debugging and confirm is env var was set correctly or not